src/AppInstallerCommonCore/AppInstallerStrings.cpp

```diff
@@ -151,6 +157,21 @@ namespace AppInstaller::Utility
         return numGraphemeClusters;
     }
 
+    size_t UTF8TerminalLength(std::string_view input)

```

Comment:
>std::string_view input [](start = 30, length = 22)

I think you might have to take in a NormalizedString to guarantee that this will work properly.  If the individual codepoints don't map to a grapheme cluster 1:1, I'm not sure what the width will report.  It is possible that there is a 0 width return value, but you don't have that represented here.

Reply:
I changed to take NormalizedString but the '...' character which takes 1 column width becomes 3 dots '.''.''.' after normalization.
So I kept string_view here and instead using NormalizedString within the function using NormalizationC (instead of the default NormalizationKC)

---
In reply to: [486834016](https://github.com/microsoft/winget-cli/pull/562#discussion_r486834016) [](ancestors = 486834016)

Reply:
Can you put tests in for the ligature fi as mentioned on [https://docs.microsoft.com/en-us/windows/win32/api/winnls/ne-winnls-norm_form](https://docs.microsoft.com/en-us/windows/win32/api/winnls/ne-winnls-norm_form)

I think that by using KC first, we have already destructed it and C won't put it back together.  Overall C should work here I think, but you might have to abandon it based on another comment.

---
In reply to: [487358887](https://github.com/microsoft/winget-cli/pull/562#discussion_r487358887) [](ancestors = 487358887,486834016)